### PR TITLE
Initialize FALCON Protocol

### PR DESCRIPTION
**Pull request by rasswanth-s**
## Description
Initialize Falcon Protocol Class.

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
Added a test for share_class

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

@LaRiffle @gmuraru @aanurraj  @kamathhrishi  Could you review my PR?

### CODE REVIEW

File: tests/sympc/protocol/falcon/falcon_test.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from sympc.protocol import FALCON
+from sympc.tensor import ReplicatedSharedTensor
+
```
Comment (kamathhrishi) [inline tests/sympc/protocol/falcon/falcon_test.py:3]: Add a test for ensuring that Falcon protocol is recognized when passed in session. A test consisting of just 
`Session(protocol="FALCON")`

File: src/sympc/tensor/mpc_tensor.py [hunk 2]
```diff
@@ -99,7 +99,6 @@ class MPCTensor(metaclass=SyMPCTensor):
         # If not, we could use the self.tensor requires_grad
         # Use for training
         "requires_grad",
-        "grad",
```
Comment (kamathhrishi) [inline src/sympc/tensor/mpc_tensor.py:102]: Was this an unintentional change?

Comment (rasswanth-s) [reply src/sympc/tensor/mpc_tensor.py:102]: There are two `grad` variables declared.It was a typo in previous PR.

Comment (kamathhrishi) [reply src/sympc/tensor/mpc_tensor.py:102]: Oh okay.

File: tests/sympc/protocol/falcon/falcon_test.py (hunk 1 above)
Comment (rasswanth-s) [reply tests/sympc/protocol/falcon/falcon_test.py:3]: Sure  @kamathhrishi

File: src/sympc/protocol/falcon/falcon.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+from typing import Any
+from typing import Dict
+from typing import List
+
+from sympc.protocol.protocol import Protocol
+from sympc.tensor import ReplicatedSharedTensor
+from sympc.tensor.tensor import SyMPCTensor
+
+
+class FALCON(metaclass=Protocol):
```
Comment (gmuraru) [inline src/sympc/protocol/falcon/falcon.py:16]: NITPICK: Could you make it snake case? (`Falcon`) - FSS looks like that because it is a short form from Function Secret Sharing (also for the Protocol name)

Comment (rasswanth-s) [reply src/sympc/protocol/falcon/falcon.py:16]: Sure @gmuraru

**Comment by rasswanth-s** [comment]
Since we have added Windows,Mac OS to our workflows.I have set  `max-parallel : 9` from `max-parallel: 4` in test.yml , to execute all our workflows simultaneously.

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -24,7 +24,7 @@ on:
 jobs:
   python-tests:
     strategy:
-      max-parallel: 4
+      max-parallel: 9
```
Comment (gmuraru) [inline .github/workflows/tests.yml:27]: Nice :+1:

**Review by gmuraru — APPROVED** [review]
LGTM! @kamathhrishi after you take a look, you can merge this :100:

**Review by kamathhrishi — APPROVED** [review]
Looks good to me. Good job.
